src/on-event/on-event.ts

```
@@ -56,38 +74,62 @@ export const createHandler = (
-      // PhysicalResourceId will change with each update, which will trigger
-      // a DELETE event for the older resource.
+
+      return {
+        PhysicalResourceId: indexName,
+        Data: { [INDEX_NAME_KEY]: indexName },
+      };
+    } else if (event.RequestType === 'Update') {
+      if (event.PhysicalResourceId == null) {
+        throw new Error('event.PhysicalResourceId is required');
+      }
+      const mapping = await getMappingFromBucket(s3, bucketParams);
+      log('Downloaded mapping from S3:', mapping);
+      await checkClusterHealth(es, maxHealthRetries);
+      log('Attempting to create index..');
+      const oldIndexName = event.PhysicalResourceId as string;
+      const newIndexName = await createIndexFromMapping(
+        es,
+        indexNamePrefix,
+        mapping
+      );
+      log(`Created index ${newIndexName}, reindexing from ${oldIndexName}..`);
+      await reIndexAllDocuments(es, oldIndexName, newIndexName);
       return {
-        PhysicalResourceId: indexId,
-        Data: {
-          [INDEX_NAME_KEY]: indexName,
-        },
+        PhysicalResourceId: newIndexName,
+        Data: { [INDEX_NAME_KEY]: newIndexName },
       };
     } else if (event.RequestType === 'Delete') {
-      const currentIndexName: string = event.ResourceProperties.IndexName;
+      if (event.PhysicalResourceId == null) {
+        throw new Error('event.PhysicalResourceId is required');
+      }
+      const currentIndexName: string = event.PhysicalResourceId;
       log(`Deleting older index: ${currentIndexName}`);
       const response = await es.indices.delete(
-        {
-          index: currentIndexName,
-        },
+        { index: currentIndexName },
         { requestTimeout: 120 * 1000, maxRetries: 0 }
       );
+
       if (response.statusCode !== 200) {
-        throw new Error();
+        throw new Error('Error when deleting the older index.');
       }
+
+      return { PhysicalResourceId: currentIndexName };
     }
+
```

At this point we shouldn't be getting unexpected event types, so consider throwing an exception here instead of returning an empty object

---

Addressed in `67ecea7`